blueflood-core/src/main/java/com/rackspacecloud/blueflood/eventemitter/RollupEventEmitter.java

```diff
@@ -56,8 +56,7 @@ public Future emit(final String event, final RollupEvent... eventPayload) {
                 @Override
                 public Future call() {
                     if (Util.shouldUseESForUnits()) {
-                        QueryDiscoveryModuleLoader.loadDiscoveryModule();
-                        final DiscoveryIO discoveryIO = QueryDiscoveryModuleLoader.getDiscoveryInstance();
+                        final DiscoveryIO discoveryIO = (DiscoveryIO) GenericClassLoader.getGenericInstance(DiscoveryIO.class, "com.rackspacecloud.blueflood.io.ElasticIO");
                         // TODO: Sync for now, but we will have to make it async eventually
```

Comment:
Modified parameters and body for method "getGenericInstance(Class c, String module)" to "getGenericInstance(Class c, CoreConfig moduleName)" and passing the arguments like "(DiscoveryIO) GenericClassLoader.getGenericInstance(DiscoveryIO.class, CoreConfig.DISCOVERY_MODULES);".